runtime/compiler/ilgen/IlGenerator.cpp

```diff
@@ -1724,6 +1724,53 @@ TR_J9ByteCodeIlGenerator::genDLTransfer(TR::Block *firstBlock)
       outmostPair = ip;
       }
 
+   // With support for unmounting virtual threads at the point of a monitorenter
+   // bytecode in Java 24 and later, the JVM requires live monitor maps produced
+   // by the JIT to list monitors that might be live correctly.  In that case,
+   // isLiveMonitorMapCorrectnessRequired() will return true.
+   //
+   // With dynamic loop transfer, the JIT might not be able to determine statically
+   // whether some monitorenter bytecodes will necessarily have been executed, and
+   // hence, will be unable to report correctly which object was the operand of
+   // each such monitorenter.
+   //
+   // Scan bytecodes leading up to the point where DLT might occur looking for any
+   // monitorenter bytecodes.  If it's possible that a monitorenter was executed
+   // prior to dynamic loop transfer from the JVM into the JIT-compiled code,
+   // fail the compilation.
+   //
+   if (fej9()->isLiveMonitorMapCorrectnessRequired() && _methodSymbol->mayContainMonitors())
+      {
+      bool mightHaveMonitorEnterBeforeDLT = false;
+
+      // Which bytecode index might the interpreter have reached in running the
+      // method prior to performing the dynamic loop transfer?  If the transfer
+      // will occur into a loop, have to consider bytecodes through to the
+      // extent of the outermost loop.
+      //
+      int32_t lastBCIndexToTest = (outmostPair != NULL) ? outmostPair->_fromIndex : dltBCIndex;
+
+      if (comp()->getOption(TR_DisableLiveMonitorMetadata))
+         {
+         mightHaveMonitorEnterBeforeDLT = true;
```

Comment:
Just to confirm, being conservative makes sense to me when we don't have live monitor metadata enabled at all, but given that the VM seems to rely on live monitor metadata to implement correct virtual thread unmounting/mounting behaviour, isn't this also applicable outside of DLT ?

Reply:
Fixed in commit c07d04de3b9e4869f5fc276d32c2f97e5c9e265c